From 06eb63699095dca38802259ebe63a7e5e231b0bf Mon Sep 17 00:00:00 2001
From: Amos King <amos.l.king@gmail.com>
Date: Thu, 29 May 2008 16:49:44 -0700
Subject: [PATCH] verify :method => :post, :redirect_to => :back
 should redirect to the referrer and not look for an action called back.

---
 actionpack/lib/action_controller/verification.rb |    2 +-
 actionpack/test/controller/verification_test.rb  |   11 +++++++++++
 2 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_controller/verification.rb b/actionpack/lib/action_controller/verification.rb
index 9f606e7..35b12a7 100644
--- a/actionpack/lib/action_controller/verification.rb
+++ b/actionpack/lib/action_controller/verification.rb
@@ -116,7 +116,7 @@ module ActionController #:nodoc:
     end
     
     def apply_redirect_to(redirect_to_option) # :nodoc:
-      redirect_to_option.is_a?(Symbol) ? self.send!(redirect_to_option) : redirect_to_option
+      (redirect_to_option.is_a?(Symbol) && redirect_to_option != :back) ? self.send!(redirect_to_option) : redirect_to_option
     end
     
     def apply_remaining_actions(options) # :nodoc:
diff --git a/actionpack/test/controller/verification_test.rb b/actionpack/test/controller/verification_test.rb
index d6fde35..f4634cf 100644
--- a/actionpack/test/controller/verification_test.rb
+++ b/actionpack/test/controller/verification_test.rb
@@ -39,6 +39,9 @@ class VerificationTest < Test::Unit::TestCase
 
     verify :only => :no_default_action, :params => "santa"
 
+		verify :only => :guarded_with_back, :method => :post,
+					 :redirect_to => :back		
+
     def guarded_one
       render :text => "#{params[:one]}"
     end
@@ -90,6 +93,10 @@ class VerificationTest < Test::Unit::TestCase
     def must_be_post
       render :text => "Was a post!"
     end
+
+		def guarded_with_back
+			render :text => "#{params[:one]}"
+		end
     
     def no_default_action
       # Will never run
@@ -109,6 +116,10 @@ class VerificationTest < Test::Unit::TestCase
     @response   = ActionController::TestResponse.new
     ActionController::Routing::Routes.add_named_route :foo, '/foo', :controller => 'test', :action => 'foo'
   end
+
+	def test_using_symbol_back_redirects_to_back
+		assert_raise(ActionController::RedirectBackError) {get :guarded_with_back}
+	end
   
   def test_no_deprecation_warning_for_named_route
     assert_not_deprecated do
-- 
1.5.5.1.316.g377d9

